Hostname alias fixes

chadswen opened the pull request:
Change the kubelet `--hostname-override` flag to use the `ansible_hostname` variable by default instead of the `inventory_hostname` specified in the inventory file. This should ensure more consistency with the value required by cloud providers like OpenStack where the kubelet hostname has to match the system hostname.

Add an `ansible_hostname` alias to `/etc/hosts` when its value is different from `inventory_hostname` to overcome node name limitations, see: [k8s issue #22770](https://github.com/kubernetes/kubernetes/issues/22770)

bogdando commented:
:+1:

mattymo commented on roles/kubernetes/preinstall/tasks/etchosts.yml:
```diff
@@ -4,7 +4,7 @@
     block: |-
       {% for item in groups['all'] -%}
-      {{ hostvars[item]['access_ip'] | default(hostvars[item]['ip'] | default(hostvars[item].ansible_default_ipv4.address)) }} {{ item }}
+      {{ hostvars[item]['access_ip'] | default(hostvars[item]['ip'] | default(hostvars[item].ansible_default_ipv4.address)) }}{% if (item != hostvars[item]['inventory_hostname_short']) %} {{ hostvars[item]['inventory_hostname_short'] }}{% endif %} {{ item }}
```
You're right! Good catch. +1

chadswen commented:
Update: moved `{{ item }}` in front of the conditionally applied alias. Changed the conditional alias to `ansible_hostname` instead of `inventory_hostname_short`, which I believe will handle the case where users set `inventory_hostname`s in their inventory file to something other than an FQDN.

mattymo commented:
+1

bogdando commented:
LGTM
